MapboxCoreNavigation/RouteTiles.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+import MapboxDirections
+
+private struct RouteTileVersion: Decodable {
+    let availableVersions: [String]
+}
+
+/**
+ `RouteTiles` object describes the meta-information about the route tiles.
+ */
+final class RouteTilesVersion {
+    
+    // MARK: Constants
+    
+    private enum Constants {
+        /// Default route tile version to use.
+        static let defaultVersion = "2020_07_03-03_00_00"
+    }
+
+    private enum UserDefaultsKey {
+        /// The key in UserDefaults where is the current route tiles version is stored.
+        static let currentRouteTilesVersion = "currentRouteTilesVersionKey"
+    }
+    
+    // MARK: Initializers
+    init(with credentials: DirectionsCredentials) {
+        directionsCredentials = credentials
+    }
+
+    // MARK: Internal properties
+    
+    /// Current version of route tiles to use.
+    var currentVersion: String {
+        get {
+            UserDefaults.standard.string(forKey: UserDefaultsKey.currentRouteTilesVersion) ?? Constants.defaultVersion
+        }
+        set {
+            UserDefaults.standard.setValue(newValue, forKey: UserDefaultsKey.currentRouteTilesVersion)
+        }
+    }
+    
+    // MARK: Private properties
+
+    private let directionsCredentials: DirectionsCredentials
+
+    /// URL to request the availalbe version of route tiles format.
+    private var requestVersionsURL: URL? {
+        let requestVersionsURLString = directionsCredentials.host.absoluteString +
+                                       "/route-tiles/v1/versions" +
+                                       "?access_token=" + (directionsCredentials.accessToken ?? "")
```

Comment:
We should create URLs using URL or URLComponents initializers rather than string concatenation or interpolation whenever possible.